packages/gatsby/src/query/query-compiler.js

```diff
@@ -113,18 +118,31 @@ class Runner {
           ),
         []
       )
+
     files = files.filter(d => !d.match(/\.d\.ts$/))
+
+    // Added this to avoid the duplicate definitions RelayParser error
+    // Worst hack ever
+    files = files.filter(d => !d.match(/\/gatsby-transformer-sharp\//))
```

Comment:
This is no-go IMO. There is plenty of plugins that provide fragments (contentful, datocms, sanity). I think we should scan all the files and then do some handling of duplicate fragments before passing them to relay compiler